handshake_client.go

```diff
@@ -624,6 +678,12 @@ func (hs *clientHandshakeState) serverResumedSession() bool {
 func (hs *clientHandshakeState) processServerHello() (bool, error) {
 	c := hs.c
 
+	// Process the delegated credential extension.
+	if hs.serverHello.delegatedCredential != nil && hs.serverHello.vers != VersionTLS12 {
```

Comment:
can you move this check to ``processDelegatedCredentialFromServer`` ?

Reply:
`processDelegatedCredentialFromServer` is called from `doTLS13Handshake` (13.go:1056), so if I move the check there, the condition must be != TLS 1.2. Shall I move it, or change the condition on handshake_client.go:682?

Reply:
I think you should move it and change.
It wouldn't be good if you fail it for TLS 1.3 I think

Reply:
In TLS 1.3, the DC is returned in the Certificate message, not the ServerHello.

Reply:
In such case:
* ``!=`` is OK, still please move the code
* comment is wrong - you don't do any processing of DC here
* in ``serverHelloMsg`` struct you need a comment explaining that ``delegatedCredential`` can be set for TLS 1.2 only

Reply:
Gotcha, I'm on it.

Reply:
Done.